lib/Connection.php

```diff
@@ -574,6 +574,9 @@ private function establishConnection() {
 	}
 
 	/**
+	 * Performs syntactic check against given host and port,
+	 * If ldapTLS disabled, the server will not be contacted!
```

Comment:
>
> So "If ldapTLS was changed on the server after the connection had been saved, the connection will fail" is true 

No, in the stated case `doConnect` method will not fail, it will return true if the host and port parameters are in the right format. This is what I am trying to explain in the comment line.

I tried to highlight the documentation note of the `ldap_connect` method: https://www.php.net/manual/en/function.ldap-connect.php . According to its documentation`ldap_connect` method is only used for checks whether the given host and port are plausible, the actual connect happens with the next calls to `ldap_*`.  In our `doConnect` method, we only make the next call if `ldapTLS` configuration is enabled. 

However, https://github.com/owncloud/user_ldap/blob/19205d77a1344d809ee77678746db6ce089cb68f/lib/Connection.php#L531 this comment assumes that, since `ldap_bind` method will run after `doConnect` method, it can only fail in the wrong password case, which is wrong. In addition, https://github.com/owncloud/user_ldap/blob/19205d77a1344d809ee77678746db6ce089cb68f/lib/LDAP.php#L319 this line makes impossible to learn why `ldap_bind` fail by disabling error code check. 

If you want, we can think of a better sentence for `doConnect` method phpdoc to explain the described behavior.

Reply:
@jvillafanez, @VicDeo Do you have any suggestions to complete this PR?

Reply:
Maybe change the doc to something like:
```
This method will perform some setup required to connect to the LDAP server,
but it won't connect to the LDAP server
```
which also makes me think if the method should throw a "ServerNotAvailableException"... probably something to review in a different PR.

Reply:
@jvillafanez the thing is if TLS enabled, it will run `ldap_start_tls` and perform a connect.

Reply:
```
This method will perform some setup required to connect to the LDAP server,
but it won't connect to the LDAP server unless ldap_start_tls is activated
```
or something along those lines

Reply:
Thank you for redirection, phpdoc updated.